File: typescript/7.整数反转.js
/*
 * @lc app=leetcode.cn id=7 lang=javascript
 *
 * [7] 整数反转
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let result = 0
    let MIN = -214748364//2 147 483 648
    let MAX = 214748364
    while (x != 0) {
        if (MIN <= result && result <= MAX)//缩小十倍比较
            result = result * 10 + x % 10
        else
            return 0
        x = ~~(x / 10)//向下取整
    }
    return result
};

// @lc code=end

File: c++/leetcode/7.整数反转.cpp
/*
 * @lc app=leetcode.cn id=7 lang=cpp
 *
 * [7] 整数反转
 */
using namespace std;
#include <limits>
// @lc code=start
class Solution {
public:
    int reverse(int x) {
        int ans = 0;
        while (x != 0) {
            if (ans < INT_MIN / 10 || ans>INT_MAX / 10)return 0;
            int digit = x % 10;
            x /= 10;
            ans = ans * 10 + digit;
        }
        return ans;
    }
};
// @lc code=end

